sdk/core/azure-core-amqp/src/impl/rust_amqp/rust_amqp/rust_wrapper/src/amqp/session.rs

```diff
@@ -255,77 +289,81 @@ mod tests {
 
     #[test]
     fn test_amqpsession_create() {
-        let session = amqpsession_create();
+        let session = unsafe { amqpsession_create() };
         assert_ne!(session, std::ptr::null_mut());
-        amqpsession_destroy(session);
+        unsafe { amqpsession_destroy(session) };
```

Comment:
Tip: you could define the whole test as `unsafe` to avoid peppering this all over if you'd rather. Fine as-is, though. Just something to consider next time to save some wear and tear on your key caps.

Reply:
Note: I only suggest for these tests. For shipping code I agree that being more specific is good, though `unsafe { }` blocks like you have below when unsafe callees are in close succession is reasonable, I think.

Reply:
Yes - These changes came from my running clippy on the tree, and there were sufficient numbers of unsafe warnings that I eventually let vs code apply the fixes automatically - I noticed I could group a ton of them together too.